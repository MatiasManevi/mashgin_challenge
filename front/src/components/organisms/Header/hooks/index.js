export { default as useCategories } from 'components/organisms/Header/hooks/useCategories';
