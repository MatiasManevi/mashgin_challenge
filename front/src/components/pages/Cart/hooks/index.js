export { default as useCart } from 'components/pages/Cart/hooks/useCart';
