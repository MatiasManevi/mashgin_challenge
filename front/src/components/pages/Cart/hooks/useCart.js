import { useState, useEffect } from 'react';


const useCart = () => {
	const [cartItems, setCartItems] = useState([]);

	const addToCart = (item) => {
		const cartItems = JSON.parse(localStorage.getItem('cart'));
		if (!cartItems) {
			const cart = {};
			cart[item.id] = item;
			localStorage.setItem('cart', JSON.stringify(cart));
			return;
		}
		cartItems[item.id] = item;
		localStorage.setItem('cart', JSON.stringify(cartItems));
	}

	const removeFromCart = (id) => {
		const cartItems = JSON.parse(localStorage.getItem('cart'));
		if (!cartItems) {
			return;
		}
		delete cartItems[id];
		localStorage.setItem('cart', JSON.stringify(cartItems));
		setCartItems(getCartItems());
	}

	const getCartItems = () => {
		const cartItems = JSON.parse(localStorage.getItem('cart'));
		if (!cartItems) return [];
		return Object.values(cartItems);
	}

	useEffect(() => {
		setCartItems(getCartItems());
	}, []);

	return {
		addToCart,
		removeFromCart,
		cartItems
	};
};

export default useCart;

