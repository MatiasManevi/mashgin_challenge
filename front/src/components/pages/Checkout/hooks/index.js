export { default as useCheckout } from 'components/pages/Checkout/hooks/useCheckout';
