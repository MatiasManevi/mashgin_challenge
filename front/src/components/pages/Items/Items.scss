@import 'assets/scss/variables/spacing';

.items {
	display: flex;
	flex-wrap: wrap;
	margin-top: $space-42;
	gap: $space-84;
	padding: $space-32;
	justify-content: center;
}
