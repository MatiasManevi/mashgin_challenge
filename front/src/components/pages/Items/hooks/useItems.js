import React from 'react';
import qs from 'qs';

import API from 'utils/apiGateway';

const useItems = () => {
	const [items, setItems] = React.useState([]);
	const search = new URLSearchParams(window.location.search).get('search');
	const category_id = new URLSearchParams(window.location.search).get('category_id');

	React.useEffect(async () => {
		const query = qs.stringify({ category_id, search });
		try {
			API.getItems(query).then(setItems);
		} catch (e) {
			console.error({ e });
		}
	}, [search, category_id]);

	return { items };
};

export default useItems;
