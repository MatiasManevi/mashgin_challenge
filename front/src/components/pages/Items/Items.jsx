import React from 'react';

import './Items.scss';
import { Item } from 'components/organisms';
import { useItems } from './hooks';

const Items = () => {
	const { items } = useItems();

	return (
		<div className="items">
			{items.map((item) => (
				<Item key={item.id} item={item} />
			))}
		</div>
	);
};

export default Items;
