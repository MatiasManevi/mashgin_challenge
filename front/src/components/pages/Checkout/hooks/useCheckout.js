import React, { useEffect, useState } from 'react';

import { CartContainer } from 'containers/Cart.container';
import API from 'utils/apiGateway';

const useCheckout = () => {
	const { getCartItems, emptyCart } = CartContainer.useContainer();
	const cartItems = getCartItems();
	const [disableSubmit, setDisableSubmit] = useState(true);
	const [submitted, setSubmitted] = useState(false);
	const [name, setName] = useState();
	const [email, setEmail] = useState();
	const [creditCard, setCreditCard] = useState();
	const total = cartItems.length ? cartItems.reduce((n, { price }) => n + price, 0) : 0;

	useEffect(() => {
		if (!email || !name || !creditCard) {
			setDisableSubmit(true);
		} else {
			setDisableSubmit(false);
		}
	}, [email, name, creditCard]);

	const submitOrder = async () => {
		const order = { name, email, creditCard, cartItems, total };
		try {
			await API.createOrder(order);
			setSubmitted(true);
			emptyCart();
		} catch (e) {
			console.error(e);
		}
	}

	return {
		setName,
		setEmail,
		setCreditCard,
		submitOrder,
		total,
		disableSubmit,
		submitted
	};
};

export default useCheckout;
