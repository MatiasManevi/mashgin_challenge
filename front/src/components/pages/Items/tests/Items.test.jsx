import React from 'react';
import { shallow } from 'enzyme';

import Items from '../Items';
import * as hooks from '../hooks';
import items from './items.fixtures';
import { ContainerProviders } from 'containers';

jest.mock('../hooks');

const stateSpy = jest.spyOn(hooks, 'useItems');

stateSpy.mockReturnValue({
	items
});

describe('Items', () => {
	it('should match snapshot', () => {
		const wrapper = shallow(<ContainerProviders><Items /></ContainerProviders>);
		expect(wrapper).toMatchSnapshot();
	});
});
