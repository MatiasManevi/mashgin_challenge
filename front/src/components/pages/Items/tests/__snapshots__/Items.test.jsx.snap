// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items should match snapshot 1`] = `
<div
  className="items"
>
  <Item
    item={
      Object {
        "condition": "new",
        "free_shipping": true,
        "id": "MLA923008999",
        "picture": "http://http2.mlstatic.com/D_712413-MLA46206646617_052021-O.jpg",
        "price": Object {
          "amount": 26399,
          "currency": "$",
          "decimals": "00",
        },
        "title": "Cacerola 24 Cm Essen Aqua Con Antiadherente ",
      }
    }
    key="MLA923008999"
  />
  <Item
    item={
      Object {
        "condition": "new",
        "free_shipping": true,
        "id": "MLA920404456",
        "picture": "http://http2.mlstatic.com/D_985487-MLA45983108078_052021-O.jpg",
        "price": Object {
          "amount": 75886,
          "currency": "$",
          "decimals": "00",
        },
        "title": "Essen Set Cacerola Sarten 24 Y Cacerola 18 Mas Utensilios",
      }
    }
    key="MLA920404456"
  />
  <Item
    item={
      Object {
        "condition": "new",
        "free_shipping": true,
        "id": "MLA839886481",
        "picture": "http://http2.mlstatic.com/D_750416-MLA44276051760_122020-O.jpg",
        "price": Object {
          "amount": 75950,
          "currency": "$",
          "decimals": "00",
        },
        "title": "Set Aqua Essen 3 Piezas + Set De Utensilios Super Oferta",
      }
    }
    key="MLA839886481"
  />
  <Item
    item={
      Object {
        "condition": "new",
        "free_shipping": true,
        "id": "MLA910856297",
        "picture": "http://http2.mlstatic.com/D_861354-MLA45116844792_032021-O.jpg",
        "price": Object {
          "amount": 62750,
          "currency": "$",
          "decimals": "00",
        },
        "title": "Essen Set Marsala 2 Piezas 28 Cm C/ant Nova Trimium Oferta!!",
      }
    }
    key="MLA910856297"
  />
</div>
`;
